src/modules/alerting.ts

```diff
@@ -32,6 +26,22 @@ const pick = <
     return newObj;
 };
 
+type ToastOptions<Type extends ValidMessageType, Timeout extends number> = {
+    /**
+     * The id of the message.  Duplicate IDs will replace each other in the toast display.  All IDs will be
+     * namespaced with your app ID to avoid collisions.  (Default will just be your app's namespace and will not support multiple messages)
+     * */
+    id?: string;
+    /** Toast type, valid options are 'error', 'info', 'success'.  (Default is 'info') */
+    type: Type;
+    /** Time in seconds to show the toast.  Set to 0 to disable automatic dismissal. (Default is 5) */
+    timeout: Type extends 'error' ? 0 : Timeout;
```

Comment:
Makes perfect sense now, but when we were chatting about this, I didn't think about the implication that type, timeout, and showCloseButton are now required.  This makes the using the options a bit clumsy as you can't just bump the timeout with `{timeout: 10}`.